content/systems/filecoin_markets/storage_market/storage_deal/storage_deal_states.md

```diff
@@ -37,4 +37,6 @@ Given deal states and their transitions, the following are the relationships bet
 - `Deal Payment`: happens on `_onSuccessfulPoSt` and at deal/sector expiration through `_submitPowerReport`, paying out `StoragePricePerEpoch` for each epoch since the last PoSt.
 - `Deal Collateral`: no storage deal collateral will be slashed for `NewDeclaredFaults` and `NewDetectedFaults` but instead some pledge collateral will be slashed given these faults' impact on consensus power. In the event of `NewTerminatedFaults`, all storage deal collateral and some pledge collateral will be slashed. Provider and client storage deal collaterals will be returned when a deal or a sector has expired. If a sector recovers from `Failing` within the `MAX_CONSECUTIVE_FAULTS` threshold, deals in that sector are still considered active. However, miners may need to top up pledge collateral when they try to `RecoverFaults` given the earlier slashing.
 
-{{<svg src="diagrams/deal-payment.mmd.svg" title="Deal States Sequence Diagram" >}}
\ No newline at end of file
+{{<svg src="diagrams/deal-payment.mmd.svg" title="Deal States Sequence Diagram" >}}
+
+{{<embed src="/modules/go-fil-markets/storagemarket/dealstatus.go"  lang="go">}}
```

Comment:
I'm very anxious about deal status going here. 

The deal status go file is internal to the markets, and refers mostly to off-chain state. The other text refers to on chain states, understood by the actor/VM. It is an ongoing source of confusion that Storage market deal status states are NOT the same as on chain states. 

Please put a header that indicates deal status go file is off-chain state tracked internally by the markets module, NOT on-chain state.